### PR TITLE
Implement XML builder module with comprehensive testing

### PR DESCRIPTION
**Pull request by michellepace**
## Summary
Implements the complete XML builder module for converting parsed Chapter objects into structured XML format with proper escaping, indentation, and template compliance.

### Key Features
• **XML Generation**: Converts `List[Chapter]` to valid XML strings using ElementTree API  
• **Template Compliance**: Exact match to SPEC.md template with proper indentation  
• **Automatic Escaping**: Handles special characters (&, ", <, >) safely  
• **Validation**: Generates parseable XML that validates with ElementTree.parse()  
• **Edge Case Support**: Handles empty chapter lists and maintains structure  

### Implementation Details
• **Pure Functions**: No side effects, easy to test and integrate  
• **Type Safety**: Full type annotations on all functions  
• **Clean Design**: Single `chapters_to_xml()` function with clear responsibility  
• **Test Coverage**: 6 focused tests with clear separation of concerns  

### Testing Results
✅ All 6 XML builder tests passing  
✅ Manual testing script validates against sample files  
✅ Generated XML matches SPEC.md template exactly  
✅ Special character escaping verified  
✅ Refactored test suite from 10→6 tests for better maintainability  

### Files Added/Modified
- `src/youtube_to_xml/xml_builder.py` - Core XML generation implementation  
- `tests/test_xml_builder.py` - 6 focused tests with elegant separation of concerns  
- `scripts/test_xml_builder_manual.py` - Manual validation script  
- `docs/SPEC.md` - Updated success criteria and added system architecture diagram  
- `CLAUDE.md` - Updated project status to reflect XML builder completion  

### Next Steps
This completes the second deliverable per the project architecture. Ready for integration with:
- `file_handler.py` (next branch: `feature/file-handler`)  
- `main.py` CLI interface  

🤖 Generated with [Claude Code](https://claude.ai/code)

Co-Authored-By: Claude <noreply@anthropic.com>

## Summary by CodeRabbit

* **New Features**
  * Added XML-building capability to generate formatted transcript XML from parsed chapters.
  * Added a manual end-to-end script to convert a transcript into XML with basic validation and progress output.

* **Documentation**
  * Rewrote project status as bullet updates; added system architecture & data-flow diagram.
  * Added handover and working notes; updated success criteria statuses.

* **Tests**
  * Added tests validating structure, escaping, indentation, and XML parseability.

* **Chores**
  * Adjusted linter ignores to permit XML parsing in tests.

### CODE REVIEW

File: docs/handovers/01-to-xml-builder.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+## Key Files in This Handover
+
+### `src/youtube_to_xml/xml_builder.py`
+- Contains `build_xml(chapters: list[Chapter]) -> ET.Element` 
+- Contains `format_xml(root: ET.Element) -> str`
+- Uses ElementTree API exclusively (per SPEC.md requirements)
+- Handles XML escaping automatically
+- Pure functions, no side effects
+
```
[the remark this answers is not in the document]
Comment (michellepace) [reply docs/handovers/01-to-xml-builder.md]: I pushed a fix in commit 7bcdadc, please review it.
